ako-gateway-api/nodes/gateway_model.go

```diff
@@ -77,8 +78,11 @@ func (o *AviObjectGraph) BuildGatewayParent(gateway *gatewayv1.Gateway, key stri
 
 func BuildPortProtocols(gateway *gatewayv1.Gateway, key string) []nodes.AviPortHostProtocol {
 	var portProtocols []nodes.AviPortHostProtocol
-	for _, listener := range gateway.Spec.Listeners {
-
+	gwStatus := akogatewayapiobjects.GatewayApiLister().GetGatewayToGatewayStatusMapping(gateway.Namespace + "/" + gateway.Name)
+	for i, listener := range gateway.Spec.Listeners {
+		if gwStatus.Listeners[i].Conditions[0].Type == string(gatewayv1.ListenerConditionAccepted) && gwStatus.Listeners[i].Conditions[0].Status == "False" {
```

Comment:
we should check if len(gwStatus.Listeners) >= i before this to avoid out of index error.  Also consider moving this check in a function since this is called in multiple places.

Reply:
During ingestion layer, while validating each and every listener, its status update is enqueued to Status layer queue. If that status update takes time or didn;t go through, we might end up processing invalid listeners.  
Should we introduce some retry if status for that listener is not there? or should we take out dependency on status update and introduce some internal structure to manage valid and invalid listeners?

Reply:
We have already introduced an internal structure, GetGatewayToGatewayStatusMapping function is taking value from that map

Reply:
Did len(gwStatus.Listeners) >= i  change